flake8_future_import.py

```diff
@@ -10,29 +10,55 @@
 except ImportError as e:
     argparse = e
 
-from ast import NodeVisitor, PyCF_ONLY_AST, Str, Module
+import ast
 
 __version__ = '0.3.2'
 
 
-class FutureImportVisitor(NodeVisitor):
+class FutureImportVisitor(ast.NodeVisitor):
 
     def __init__(self):
         super(FutureImportVisitor, self).__init__()
         self.future_imports = []
+
         self._uses_code = False
+        self._uses_print = False
+        self._uses_division = False
+        self._uses_import = False
+        self._uses_str_literals = False
+
+    def _is_print(self, node):
+        # python 2
+        if hasattr(ast, 'Print') and isinstance(node, ast.Print):
+            return True
+
+        # python 3
+        if isinstance(node, ast.Call) and \
+           isinstance(node.func, ast.Name) and \
+           node.func.id == 'print':
+            return True
+
+        return False
 
     def visit_ImportFrom(self, node):
         if node.module == '__future__':
             self.future_imports += [node]
-
-    def visit_Expr(self, node):
-        if not isinstance(node.value, Str) or node.value.col_offset != 0:
-            self._uses_code = True
+        else:
+            self._uses_import = True
 
     def generic_visit(self, node):
-        if not isinstance(node, Module):
+        if not isinstance(node, ast.Module):
             self._uses_code = True
+
+        if isinstance(node, ast.Str):
```

Comment:
for absolute correctness, `unicode_literals` should only be required when there are string literals that are not prefixed ... i.e. if a module uses b'..' throughout, `unicode_literals` has no effect.

This may also need ast.FormattedValue and ast.JoinedStr, on Python 3.6 -- i havent tested these.